modules/ui/org.eclipse.fx.ui.preferences/src/main/java/org/eclipse/fx/ui/preferences/page/FieldEditorPreferencePage.java

```diff
@@ -35,91 +44,140 @@
 		this.parent.setCenter(grid);
 		this.parent.setBottom(actions);
 		actions.setAlignment(Pos.BASELINE_RIGHT);
-		
+
 		grid.getStyleClass().add(PAGE_GRID_STYLE);
-		actions.getStyleClass().add(PAGE_ACTIONS_STYLE);		
+		actions.getStyleClass().add(PAGE_ACTIONS_STYLE);
 		actions.setMinWidth(Region.USE_PREF_SIZE);
 		grid.setMinWidth(Region.USE_PREF_SIZE);
 		grid.setMinHeight(Region.USE_PREF_SIZE);
 	}
-	
+
 	protected String getUserAgentStylesheet() {
 		return null;
 	}
 
 	@Override
 	public Command<Void> persist() {
-		return Command.createCommand(() -> editors.forEach(FieldEditor::persist));
+		return persistCommand;
 	}
-	
+
 	@Override
 	public Optional<Command<Void>> restoreDefault() {
 		return Optional.of(Command.createCommand(() -> {
 			editors.forEach(FieldEditor::restoreDefaults);
 		}));
 	}
-	
+
 	@PostConstruct
 	void initFieldEditors() {
-		createActionButtons();
 		createFieldEditors();
+		createActionButtons();
 		bindErrorMessages();
 		reset().execute();
 	}
-	
+
 	private void bindErrorMessages() {
-		List<StringExpression> errorMessages = editors.stream().map(FieldEditor::errorMessage).collect(Collectors.toList());
-		StringExpression concat = FXBindings.concat("\n", errorMessages.toArray(new ObservableValue[errorMessages.size()]));
-		concat.addListener((obs, old, newV) -> System.out.println("'"+newV+"'"));
-		this.errorMessage.textProperty().bind(concat);
+		ObservableValue<Status> pageStatus = getPageStatus();
+		
+		this.errorMessage.textProperty().bind(Bindings.createStringBinding(() -> pageStatus.getValue().getMessage(), pageStatus));
 		this.errorMessage.managedProperty().bind(this.errorMessage.visibleProperty());
 		this.errorMessage.visibleProperty().bind(this.errorMessage.textProperty().isNotEmpty());
+		
+		FXObservableUtil.onChange(pageStatus, status -> {
+			boolean isWarning = status.getState() == State.WARNING;
+			boolean isError = status.getState() == State.ERROR;
+			
+			this.errorMessage.pseudoClassStateChanged(WARNING_STYLE, isWarning);
+			this.errorMessage.pseudoClassStateChanged(ERROR_STYLE, isError);
+		});
 	}
 
-	abstract protected void createFieldEditors();
+	private ObservableValue<Status> getPageStatus(){
+		List<ObservableValue<Status>> allStatuses = getAllStatuses();
+		
+		Callable<Status> toMultistatus = () -> allStatuses.stream()
+				.map(ObservableValue::getValue)
+				.filter(s -> s != null && ! s.isOk())
+				.collect(MultiStatus.toMultiStatus(getMessage(allStatuses), Status.UNKNOWN_RETURN_CODE));
+		
+		return Bindings.createObjectBinding(toMultistatus, allStatuses.toArray(new ObservableValue<?>[0]));
+	}
 	
+	private List<ObservableValue<Status>> getAllStatuses() {
+		return editors.stream()
+				.map(FieldEditor::statusProperty)
+				.collect(Collectors.toList());
+	}
+
+	private String getMessage(List<ObservableValue<Status>> allStatuses) {
+		return allStatuses.stream()
+			.map(ObservableValue::getValue)
+			.filter(s -> s != null && !s.isOk()) 
```

Comment:
if you split the null and isOk check you can use pure method references `.filter(Objects::nonNull).filter(Status::isOK)`

Reply:
Unfortunately, we're retaining "non-OK" statuses, so I can't use the Status::isOK reference :) (And I believe there is no simple way to negate method reference predicates)

Reply:
You are absolutely right (you have to cast it to Predicate and then call negate i guess) but what about adding a isNotOk()-method to Status?